[SALAD-24633] Add initial functionality (#1)
SaladTechnologies/salad-cloud-transcription-sdk-javascript: README.md

## Versions

- API version: `0.9.0-alpha.1`

> - SDK version: `0.9.0-alpha.2`

Reviewer: We should leave this at `alpha.1`- we haven't released any versions to npm yet.

npm install @saladtechnologies-oss/salad-cloud-transcription-sdk
```

## Authentication

### API Key Authentication

> The SaladCloudTranscriptionSdk API uses API keys as a form of authentication. An API key is a unique identifier used to authenticate a user, developer, or a program that is calling the API.

Reviewer: ```suggestion
The SaladCloud Transcription SDK uses API keys as a form of authentication. An API key is a unique identifier used to authenticate a user, developer, or a program that is calling the API.
```

#### Setting the API key

When you initialize the SDK, you can set the API key as follows:

```ts

> const saladCloudTranscriptionSdk = new SaladCloudTranscriptionSdk({ apiKey: 'YOUR_API_KEY' })

Reviewer: Let's keep the sample variable names simple so the lines are shorter...

```suggestion
const sdk = new SaladCloudTranscriptionSdk({ apiKey: 'YOUR_API_KEY' })
```

```

## Setting a Custom Timeout

You can set a custom timeout for the SDK's HTTP requests as follows:

```ts

> const saladCloudTranscriptionSdk = new SaladCloudTranscriptionSdk({ timeout: 10000 })

Reviewer: ```suggestion
const sdk = new SaladCloudTranscriptionSdk({ timeout: 10000 })
```

```

# Sample Usage

Below is a comprehensive example demonstrating how to authenticate and transcribe:

```ts

> import { SaladCloudTranscriptionSdk } from '@saladtechnologies-oss/salad-cloud-transcription-sdk'
> ;(async () => {
>   const saladCloudTranscriptionSdk = new SaladCloudTranscriptionSdk({
>     apiKey: 'YOUR_API_KEY',
>   })
> 
>   const { id } = await saladCloudTranscriptionSdk.transcribe('organization_name', 'path_to_file/video.mp4')
> 
>   console.log(id)
> })()

Reviewer: ```suggestion
import { SaladCloudTranscriptionSdk } from '@saladtechnologies-oss/salad-cloud-transcription-sdk';

(async () => {
  const sdk = new SaladCloudTranscriptionSdk({
    apiKey: 'YOUR_API_KEY',
  })

  const { id } = await sdk.transcribe('organization_name', 'path_to_file/video.mp4')

  console.log(id)
})()
```

npm install @saladtechnologies-oss/salad-cloud-transcription-sdk
```

## Authentication

### API Key Authentication

The SaladCloud Transcription SDK uses API keys as a form of authentication. An API key is a unique identifier used to authenticate a user, developer, or a program that is calling the API.

#### Setting the API key

When you initialize the SDK, you can set the API key as follows:

```ts
const sdk = new SaladCloudTranscriptionSdk({ apiKey: 'YOUR_API_KEY' })
```

## Setting a Custom Timeout

You can set a custom timeout for the SDK's HTTP requests as follows:

```ts
const sdk = new SaladCloudTranscriptionSdk({ timeout: 10000 })
```

# Sample Usage

Below is a comprehensive example demonstrating how to authenticate and transcribe:

```ts
import { SaladCloudTranscriptionSdk } from '@saladtechnologies-oss/salad-cloud-transcription-sdk'

> ;(async () => {

Reviewer: What is happening with the formatting here?

Author: The semicolon before IIFE is a defensive [coding practice](https://thedavidbarton.github.io/blog/anonymous-IIFE-fn/). In JavaScript, if the preceding code doesn’t end with a semicolon, the parser might mistakenly combine the previous statement with the IIFE, leading to errors.

I can change it to
```
import { SaladCloudTranscriptionSdk } from '@saladtechnologies-oss/salad-cloud-transcription-sdk'

const transcribe = async () => {
  const sdk = new SaladCloudTranscriptionSdk({
    apiKey: 'YOUR_API_KEY',
  })

  const { id } = await sdk.transcribe('organization_name', 'path_to_file/video.mp4')
  return id
}

transcribe()
```

Author: I will change Readme in the following PR after adding example folder. So it aligns with the actual example.

Reviewer: Yes, so why isn't the semicolon on the preceding line?

```ts
import { SaladCloudTranscriptionSdk } from '@saladtechnologies-oss/salad-cloud-transcription-sdk';
```

I don't think this example should be formatted like this. Check the examples in the README on the base SDK:

https://github.com/SaladTechnologies/salad-cloud-sdk-javascript

Author: Changed it

    const sdk = new SaladCloudTranscriptionSdk({
        apiKey: 'YOUR_API_KEY',
    })

    const { id } = await saladCloudTranscriptionSdk.transcribe('organization_name', 'path_to_file/video.mp4')

>     console.log(id)

Reviewer 2: i know this is just the readme but should this be a console.log?

Author: Since the ID appears in the console, I think it's valuable to keep it for clarity.
